Substitute words loaded from separate file, new additions

belliriccardo opened the pull request:
Made it such that new additions can just be appended at the end of the `.csv` file to keep the main file more clean. Also, there are plenty more substitutions. Arrr.

belliriccardo committed:
Substitute words from separate file, new additions

ntoll commented:
Ahoy there @belliriccardo,

Welcome to the crew. :skull_and_crossbones: 

Ye have landed plenty'o shiny phrases in that there CSV.

But surely it should be a CSR file..? Ye say it, "Sea, yes! Arrr..!" (but I jest).

This'll need fixing up before we sail it to the PyPI lagoons. Anyone doing `pip install arrr` will need a treasure map to find the missing CSV file. Savvy..?

Summat for ye to patch up before we hoist this sail..? But happy to accept your contribution to the crew when patched.

Arrrrrr!

:pirate_flag:

belliriccardo commented:
Ye're right matey, but I be missin' th' skill to package this altogether. If ye need me, I'll be in th' quarters below deck, cap'n...

ntoll commented:
Aye. I'll keel haul this out in the coming week or so. :+1:

belliriccardo committed:
Add csv file in module

belliriccardo commented:
Disregard me last pigeon message, I 'ave figured it out! Should be jolly an' ready to go.

belliriccardo committed:
Add csv file in module

belliriccardo committed:
New words + Arrr misbehaviour with no arrrguments

belliriccardo committed:
New words

belliriccardo committed:
Make it such that words that end with punctuation get translated correctly. Example:
Before: "How are you?" -> "How be you?"
Now: "How are you?" -> "How be ye?"
Also don't de-capitalize words in the main dict so that names don't get lowercase'd (Ex. "Jack" -> "jack" inside a string of text no longer happens)

ntoll commented:
Hmm... I've just tested it and it doesn't work. I think I know why, and will fix/check/merge when done. :+1:

merwok commented on arrr.py:
```diff
@@ -24,66 +24,25 @@
 CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 """
 import argparse as arrrgparse  # Geddit..? ;-)
+from csv import reader as HELMSMAN  # He be the only lad aboard who can read!
```
Why are ye screaming lads?  `helmsman` be the good and usual way.

merwok commented on arrr.py:
```diff
@@ -24,66 +24,25 @@
 #: MAJOR, MINOR, RELEASE, STATUS [alpha, beta, final], VERSION
-_VERSION = (1, 0, 3)
-
+_VERSION = (1, 1, 0)
```
Your editor be removing all them nice blank lines.  Mates need restin betwen workin, ya know?

merwok commented on arrr.py:
```diff
@@ -24,66 +24,25 @@
-}
-
+_PIRATE_WORDS = dict()
+with open('BLABBER.csv') as NONSENSE:
```
This here data file needs to be in yon package there mate, and ye be better usin pkgutil or importlib.resources to access it in a portable way!

merwok commented on setup.py:
```diff
@@ -31,5 +31,6 @@
         "Topic :: Communications",
         "Topic :: Software Development :: Internationalization",
     ],
+    include_package_data=True,
```
This here line is good but does nothin’ for now: yon CSV trove must be inside yon package to be included as package booty!

belliriccardo commented:
Ahoy @merwok ! Me've forgotten a-boot(harharhar) this 'ere galley. Me've been' havin' a not so jolly time doin' me last pirate exams. About that there CSV file, I 'ave added it to th' MANIFEST.in file, I was thinkin' that was what was needed. If ye have any more comments or if I'm missin' somethin', feel free to tell! I'll go walk the plank now

merwok commented:
Nah laddie, yon manifest is for loading extra cargo say readme and such into yon sdist barrel, but that’s not gonna install them there files!  Trust me old head and move yon file into yon package, that’s what ye need to get it packaged and installed and accessed, yarr!

ntoll commented:
Ahoy there mateys..!

@merwok you be an old sea dog full of packaging tales and lore for getting the booty in th'right place? Perchance you'd scrawl a treasure map for crew mate @belliriccardo to bury his CSV treasure in the right place? Yarrrrr...

Rum and a rebase may be needed too, before this plan can happen (but it sounds like a profitable piratical enterprise).

Commit:
turn module into package, fix packaging

merwok commented:
This here chart shows the way to get what I be sayin: https://github.com/belliriccardo/arrr/pull/1

ntoll commented:
Ahoy there me maties..! Merry Christmas and wishing you good plunderin' in the new year. Yarrrr.

If @belliriccardo lets @merwok's on board, I'll make it walk the plank into the shark infested waters o'this project.

belliriccardo committed:
Merge pull request #1 from merwok-forks/feature/packaging

turn module into package, fix packaging

belliriccardo commented:
Allright! Sorry for me absence, ya know me've been plunderin' all these holidays! I 'ave merged @merwok pull request, sould be jolly now!

merwok commented on arrr/__init__.py:
```diff
@@ -35,55 +33,16 @@
 """
 
-
 #: MAJOR, MINOR, RELEASE, STATUS [alpha, beta, final], VERSION
```
```suggestion

#: MAJOR, MINOR, RELEASE, STATUS [alpha, beta, final], VERSION
```

merwok commented on arrr/__init__.py:
```diff
@@ -35,55 +33,16 @@
 #: MAJOR, MINOR, RELEASE, STATUS [alpha, beta, final], VERSION
-_VERSION = (1, 0, 3)
-
+_VERSION = (1, 1, 0)
```
```suggestion

_VERSION = (1, 1, 0)
```

merwok commented on arrr/__init__.py:
```diff
@@ -35,55 +33,16 @@
 
+del nonsense, gibberish
 
 #: A list of Pirate phrases to randomly insert before or after sentences.
```
```suggestion

#: A list of Pirate phrases to randomly insert before or after sentences.
```

merwok commented on arrr/__init__.py:
```diff
@@ -161,6 +128,10 @@ def main(arrrgv=None):
                 "Summat went awry, me lovely! Arrr..."
             )
```
```suggestion
                "Summat went awry, me lovely! Arrr...",
                file=sys.stderr,
            )
```

merwok replied:
Ye should complain on tha stderr room laddie!

merwok commented on arrr/__init__.py:
```diff
@@ -161,6 +128,10 @@ def main(arrrgv=None):
             sys.exit(1)
+    else:
+        print(
+            "Ye filthy bilge rat, don't try to fool me! I will gut yer insides!"
+        )
```
```suggestion
            "Ye filthy bilge rat, don't try to fool me! I will gut yer insides!",
            file=sys.stderr,
        )
        sys.exit(1)
```

merwok commented on arrr/blabber.csv:
```diff
@@ -0,0 +1,253 @@
+you're,ye be
+you,ye
+your,yer
+yourself,yerself
```
```suggestion
yourself,yerself
```

merwok commented on arrr/__init__.py:
```diff
@@ -35,55 +33,16 @@
-    "meeting": "parley with rum and cap'n",
-}
+_PIRATE_WORDS = dict()
+with open('BLABBER.csv') as nonsense:
```
Python landlubbers think it good practiss to eschew side effects at importin’ time, like openin’ and readin’ file.
That’s fine thinkin’ but it requires ye to go roundabout and add some `init` function like this here: https://github.com/python/cpython/blob/main/Lib/mimetypes.py#L336-L337

So me wonder what cap’n be thinkin’.  Be fine to keep this simple if `arrr` be mainly a command-line program, not a module for other landlubbers to use.

merwok reviewed (commented):
yo ho ho
